test/Conversion/MooreToCore/basic.mlir

```diff
@@ -1097,3 +1097,16 @@ moore.module @scfInsideProcess(in %in0: !moore.i32, in %in1: !moore.i32) {
     moore.return
   }
 }
+
+// CHECK-LABEL: @blockArgAsObservedValue
+moore.module @blockArgAsObservedValue(in %in0: !moore.i32, in %in1: !moore.i32) {
+  %var = moore.variable : <!moore.i32>
+  moore.procedure always_comb {
+    // CHECK: [[COND:%.+]] = comb.icmp slt %in1, %{{.*}} : i32
+    // CHECK: [[IN0:%.+]] = comb.mux [[COND]], %{{.*}}, %in0 : i32
+    // CHECK: [[IN1:%.+]] = comb.mux [[COND]], %{{.*}}, %in1 : i32
```

Comment:
Nit: it might be good to check for the `llhd.wait` op and the values that got added to its `observed` list.